internal/discovery/namespace_cache.go

```diff
@@ -0,0 +1,90 @@
+package discovery
+
+import (
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// DefaultTTL is default ttl of the cache entries.
+	DefaultTTL = 10 * time.Minute
```

Comment:
This is not used, is it?

Reply:
Removed this as well as the struct field, they were from the old cache. Done in 41e75891a250c10f062d391f5218e08850cb941c.